activerecord/lib/active_record/transactions.rb

```diff
@@ -292,9 +292,10 @@ def set_options_for_callbacks!(args, enforced_options = {})
 
           if options[:on]
             fire_on = Array(options[:on])
+            idempotent = options[:idempotent]
```

Comment:
Perhaps it's just because I'm not a native english speaker, but `idempotent` sounds needlessly technical to me. Like we're barfing out SQL intricacies to end users.

It could just be an unfamiliarity with the word, though I've been trying to read it several times and it still doesn't click.

`consider_succeeded: true`, might be closer to the SQL-forces-us-to-consider-this-final aspect, though still needs finagling.